packages/smart-contracts/src/contracts/ERC20SwapToConversion.sol

```diff
@@ -57,104 +58,111 @@ contract ERC20SwapToConversion is Ownable {
   * @param _uniswapDeadline Deadline for the swap to be valid
   * @param _chainlinkMaxRateTimespan Max time span with the oldestrate, ignored if zero
   */
-  function swapTransferWithReference(
-    address _to,
-    uint256 _requestAmount, // requestCurrency
-    uint256 _amountInMax, // SpentToken
-    address[] memory _uniswapPath, // from paymentNetworkToken to spentToken on uniswap
-    address[] memory _chainlinkPath, // from requestCurrency to spentToken on chainlink
-    bytes calldata _paymentReference,
-    uint256 _requestFeeAmount, // requestCurrency
-    address _feeAddress,
-    uint256 _uniswapDeadline,
-    uint256 _chainlinkMaxRateTimespan
-  )
-    external
-  {
-    require(
-      _uniswapPath[_uniswapPath.length-1] == _chainlinkPath[_chainlinkPath.length-1],
-      "the requested token on uniswap must be the payment currency");
-
-    // Get the amount to pay in paymentNetworkToken
-    uint256 paymentNetworkTotalAmount = getConversion(_chainlinkPath, _requestAmount, _requestFeeAmount);
-
-    require(
-      IERC20(_uniswapPath[0]).safeTransferFrom(msg.sender, address(this), _amountInMax),
-      "Could not transfer payment token from swapper-payer");
-
-    swapRouter.swapTokensForExactTokens(
-      paymentNetworkTotalAmount,
-      _amountInMax,
-      _uniswapPath,
-      address(this),
-      _uniswapDeadline
-    );
-
-    // Pay the request and fees
-    paymentProxy.transferFromWithReferenceAndFee(
-      _to,
-      _requestAmount,
-      _chainlinkPath,
-      _paymentReference,
-      _requestFeeAmount,
-      _feeAddress,
-      paymentNetworkTotalAmount, // _maxToSpend
-      _chainlinkMaxRateTimespan
-    );
-
-    // Give the change back to the payer, in both currencies (only spent token should remain)
-    if (IERC20(_uniswapPath[0]).balanceOf(address(this)) > 0) {
-      IERC20(_uniswapPath[0]).safeTransfer(msg.sender, IERC20(_uniswapPath[0]).balanceOf(address(this)));
+    function swapTransferWithReference(
+        address _paymentProxy,
+        address _to,
+        uint256 _requestAmount, // requestCurrency
+        uint256 _amountInMax, // SpentToken
+        address[] memory _uniswapPath, // from spentToken to expectedToken on uniswap
+        address[] memory _chainlinkPath, // from invoicingCurrency to expectedToken on chainlink
+        bytes memory _paymentReference,
```

Comment:
Switch from calldata to memory to free memory in the stack. Reaching the 16 variables limit otherwise.